OpenEdXMobile/src/main/java/org/edx/mobile/view/EditUserProfileFragment.java

```diff
@@ -514,4 +528,90 @@ private static TextView createField(@NonNull LayoutInflater inflater, @NonNull V
         parent.addView(textView);
         return textView;
     }
+
+    public boolean checkPermissions(int requestCode) {
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
+
+            switch (requestCode) {
+                case CAMERA_PERMISSION_REQUEST:
+                    if (ContextCompat.checkSelfPermission(getActivity(), Manifest.permission.CAMERA) !=
+                            PackageManager.PERMISSION_GRANTED) {
+                        return false;
+                    }
+                    break;
+                case READ_STORAGE_PERMISSION_REQUEST:
+                    if (ContextCompat.checkSelfPermission(getActivity(), Manifest.permission.READ_EXTERNAL_STORAGE) !=
+                            PackageManager.PERMISSION_GRANTED) {
+                        return false;
+                    }
+                    break;
+
+                default:
+                    break;
+            }
+        }
+        return true;
+    }
+
+    @TargetApi(Build.VERSION_CODES.M)
+    public void requestPermissions(int requestCode) {
```

Comment:
This function and `checkPermissions` function can be moved into a util class e.g. `PermissionsUtil` to standardize permissions usage throughout the app.